packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -623,12 +623,13 @@ export class DataStores implements IDisposable {
     }
 
     /**
-     * When running GC in test mode, this is called to delete objects whose routes are unused. This enables testing
-     * scenarios with accessing deleted content.
+     * This is called to update objects whose routes are unused. The unused objects are either deleted or marked as
+     * tombstones.
      * @param unusedRoutes - The routes that are unused in all data stores in this Container.
-     * @param tombstone - set the objects corresponding to routes as tombstones.
+     * @param tombstone - if true, the objects corresponding to unused routes are marked tombstones. Otherwise, they
+     * are deleted.
      */
-    public deleteUnusedRoutes(unusedRoutes: string[], tombstone: boolean = false) {
+    public updateUnusedRoutes(unusedRoutes: string[], tombstone: boolean) {
```

Comment:
While we're at it, we should be failing fast if pathParts.length != 2 down below. If I pass in a blob route here, we should fail, and if I pass in a dds route in here, we should also fail. If we want to add logic for DDS's we can do that later :)

Reply:
We already assert if we don't find a context with the id so the blob route part is taken care of. You can still call it for DDS, it would just be a no-op (in test mode, it is called for DDS too). We can change it so that its callled only for data stores but I don't see any advatages of doing that.

Reply:
This is a non-blocker, I'll leave the choice up to you.